$OpenBSD: patch-src_ThreadTasks_cpp,v 1.1 2012/10/25 07:59:45 dcoppa Exp $

Fix memory leak when AICH hashing already known files
(upstream git commit 9e62350fae9f24de64987a0cb002fdc15b5fa9af)

--- src/ThreadTasks.cpp.orig	Sat Aug 27 22:28:32 2011
+++ src/ThreadTasks.cpp	Thu Oct 25 11:09:25 2012
@@ -189,6 +189,8 @@ void CHashingTask::Entry()
 					CFormat(wxT("Warning, failed to save AICH hashset for file: %s"))
 						% m_filename );
 			}
+			// delete hashset now to free memory
+			AICHHashSet->FreeHashSet();
 		}
 	}
 	
